src/main/java/com/libraryproject/librarysystem/service/impl/BookServiceImplementation.java

```diff
@@ -0,0 +1,35 @@
+package com.libraryproject.librarysystem.service.impl;
+
+import com.libraryproject.librarysystem.dto.BookResponseDto;
+import com.libraryproject.librarysystem.entity.Book;
+import com.libraryproject.librarysystem.mapper.BookMapper;
+import com.libraryproject.librarysystem.repository.BookRepository;
+import com.libraryproject.librarysystem.service.BookService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class BookServiceImplementation implements BookService {
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+    private static final int PAGE_SIZE = 10;
+
+    @Override
+    public Page<BookResponseDto> findAllBooksWithPagination(int pageNumber) {
+        Page<Book> bookPage = bookRepository.findAll(PageRequest.of(pageNumber, PAGE_SIZE));
+        Page<BookResponseDto> resultPage = bookPage.map(bookMapper::mapBookToBookResponseDto);
+        return resultPage;
+    }
+
+    @Override
+    public List<BookResponseDto> findAllBooks() {
+        List<Book> books = bookRepository.findAll();
+        return bookMapper.mapBooksToBookResponseDtos(books);
```

Comment:
Pytasz czy powinno mapować się pojedyncze elementy czy całą listę... Oba podejścia są poprawne, możnaby ewentualnie pokusić się o analizę kodu, pod kątem które rozwiązanie jest bardziej optymalne. 
Patrząc na implementację mappera (target/generated-sources/annotations/com/libraryproject/librarysystem/mapper/BookMapperImpl.java) 
![image](https://github.com/AleksandraWlazlo/Library-System/assets/22994854/cd2416ac-f92b-4363-9fb8-d2c8ff8ab587)

okazuje się że nie ma to większego znaczenia, ponieważ mapperListy wywołuje pojedynczy mapper w pętlli forEach. (Moooooże użycie parallerStream byłoby bardziej poprawne optymalizacyjnie, ale nie jest to zysk warty w tej chwili zachodu :) )

Reply:
Dzięki za info. ParallerStream - nie wiem co to, jutro poczytam